src/pefile.cpp

```diff
@@ -1825,9 +1830,10 @@ void PeFile::processResources(Resource *res,unsigned newaddr)
     while (res->next())
         if (res->newoffs())
             res->newoffs() += newaddr;
-    upx_byte *p = res->build();
-    memcpy(oresources,p,res->dirsize());
-    delete [] p;
+    if (res->dirsize()) {
```

Comment:
note: previous calls to new() would be ok for size of 0, but MemBuffer asserts size > 0. To avoid failing this assert, we now don't build() when dirsize() == 0.